specs/WebRtcPortConfiguration.md

```diff
@@ -0,0 +1,195 @@
+- Avoid conflicts with other applications that may already use ephemeral ranges.  
+
+Usage steps:  
+1. Create `CoreWebView2EnvironmentOptions`.  
+2. Access the `WebRtcPortConfiguration` object.  
+3. Call `SetPortRange` for `CoreWebView2WebRtcProtocolKind.Udp`.  
+4. Pass the options when creating the WebView2 environment.  
+
+
+# Examples
+### C++ Configure UDP Port Range
+```cpp
+wil::com_ptr<ICoreWebView2EnvironmentOptions> options = 
+    Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+
+wil::com_ptr<ICoreWebView2WebRtcPortConfiguration> portConfig;
+CHECK_FAILURE(options->get_WebRtcPortConfiguration(&portConfig));
+
+CHECK_FAILURE(portConfig->SetPortRange(
+    CoreWebView2WebRtcProtocolKind::Udp, 50000, 51000));
+
+HRESULT hr = CreateCoreWebView2EnvironmentWithOptions(
+        subFolder, m_userDataFolder.c_str(), options.Get(),
+        Callback<ICoreWebView2CreateCoreWebView2EnvironmentCompletedHandler>(
+            this, &AppWindow::OnCreateEnvironmentCompleted)
+            .Get());
+```
+
+### C# Configure UDP Port Range
+```csharp
+var options = new CoreWebView2EnvironmentOptions();
+
+var portConfig = options.WebRtcPortConfiguration;
+portConfig.SetPortRange(CoreWebView2WebRtcProtocolKind.Udp, 50000, 51000);
+
+var environment = await CoreWebView2Environment.CreateAsync(
+    browserExecutableFolder: subFolder,
+    userDataFolder: m_userDataFolder,
+    options: options);
+
+OnCreateEnvironmentCompleted(environment);
+```
+
+# API Details
+### C++  
+```
+/// Additional options used to create WebView2 Environment to manage WebRTC UDP port range configuration.
+[uuid(0eebe393-8dcf-5bc5-a15d-d862088242e9), object, pointer_default(unique)]
+interface ICoreWebView2EnvironmentOptions10 : IUnknown {
+  /// Get the WebRTC port range configuration object for configuring a custom port range.
+  /// This configuration object can be used to set and retrieve port range configuration
+  /// that WebRTC will use for ICE candidates and media connections.
+  /// If no custom range is configured, WebRTC will use the operating system's default dynamic port range.
+  /// Configuration must be completed before the environment is created. Once the environment is
+  /// created, the port range cannot be customized.
+  /// 
+  /// 
+  /// \snippet AppWindow.cpp WebRtcPortConfiguration
+  [propget] HRESULT WebRtcPortConfiguration([out, retval] ICoreWebView2WebRtcPortConfiguration** value);
+}
+
+/// WebRTC port configuration interface for managing WebRTC port range configuration.
+/// This interface provides methods to configure and retrieve custom port ranges
+/// that WebRTC will use for ICE candidates and media connections across different protocols.
+[uuid(b1ac2eb4-15b5-574f-aeb7-c51b9f1520fa), object, pointer_default(unique)]
+interface ICoreWebView2WebRtcPortConfiguration : IUnknown {
+  /// The `SetPortRange` method allows you to set a custom port range for WebRTC to use
```

Comment:
Confirming, there is no case the caller would not be able to set the portrange, that the caller should handle?

(Exceptions for bad args, or internal catastrophic failures.  There are no execution failures.  

For example, having access to CoreWebView2EnvironmentOptions means I'm always allowed to set this range.  Does setting the range on the Options object guarantee the range takes effect on the CoreWebView2Environment?  

Could a portrang result in CoreWebView2Environment::CreateWithOptionsAsync failing?)

Reply:
As long as you meet the requirements listed here in the docs you should be good. There may be additional WebRTC failures but WebView2 should work.